src/parcel_snoopi_deep.jl

```diff
@@ -12,6 +12,8 @@ const flamegraph = FlameGraphs.flamegraph  # For re-export
 Flatten the execution graph of Timings returned from `@snoopi_deep` into a Vector of pairs,
 with the exclusive time for each invocation of type inference, skipping any frames that
 took less than `tmin_secs` seconds. Results are sorted by time.
+
+`ROOT` is a dummy element whose time corresponds to the total time for the operation, not just inference time.
```

Comment:
Ah, actually, `ROOT` is the _complement to_ inference time. It's the total time _minus_ inference time. It's like the "exclusive time" for the top-level node.

Sorry if that's confusing!

Reply:
It looks like the only place I've documented that is here, and it's pretty poor:
https://github.com/JuliaLang/julia/blob/aee4c26ae4e5c3209a0a4962de813a34f78d5c9c/base/compiler/typeinfer.jl#L73-L83

Thanks for adding this line to these functions! 👍 Good idea

Reply:
Maybe something like this would clarify things?:

```suggestion
`ROOT` is a dummy element whose time corresponds to the sum of time spent outside inference. It's
the total time of the operation minus the total time for inference. You can run `sum(first.(result[1:end-1]))`
to get the total inference time, and `sum(first.(result))` to get the total time overall.
```

Reply:
Also also, i've been wanting to redo the flamegraph a bit to actually put the `ROOT()` nodes back into the flamegraph, inbetween the inference nodes, which I think will help clarify this? :)

Reply:
> ROOT is the complement to inference time

Thanks. I got this correct in the blog post but I must not have noticed/remembered this yet when creating this PR. I like your suggested change.

Reply:
> actually put the ROOT() nodes back into the flamegraph

That does seem like it would help, though with the combination of my 3 PRs I now rarely look at the flamegraph. That might be for reasons of performance though, and you've fixed that, so this seems worth having.

Reply:
> I now rarely look at the flamegraph. That might be for reasons of performance though, and you've fixed that, so this seems worth having.

Also, if you haven't already tried it, I'd _very much_ recommend trying to display the flame graph in PProf.jl, via `pprof(fg)`. In my opinion it's a _really_ nice experience to see all of the method instances laid out with names visible, and it allows searching and filtering, all of which really help with digging into performance.

Also, the graph view is useful if we support aggregated views, as is done in #160. :)

Reply:
:) I've opened a PR to add PProf as one of the recommended viewers for FlameGraphs, since i realized it wasn't there yet! :)
https://github.com/timholy/FlameGraphs.jl/pull/32